src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php

```diff
@@ -68,18 +68,29 @@ public function lateCollect()
         if (null !== $this->logger) {
             $containerDeprecationLogs = $this->getContainerDeprecationLogs();
             $this->data = $this->computeErrorsCount($containerDeprecationLogs);
+            $this->data['deprecation_logs'] = $containerDeprecationLogs;
             // get compiler logs later (only when they are needed) to improve performance
             $this->data['compiler_logs'] = [];
             $this->data['compiler_logs_filepath'] = $this->containerPathPrefix.'Compiler.log';
-            $this->data['logs'] = $this->sanitizeLogs(array_merge($this->logger->getLogs($this->currentRequest), $containerDeprecationLogs));
-            $this->data = $this->cloneVar($this->data);
+            $this->data['logs'] = $this->logger->getLogs($this->currentRequest);
+            $this->data['logs_are_processed'] = false;
```

Comment:
Micro optimization: You could not set the value and in getLogs you check for `!isset`.

Reply:
I think it's better to define the option. Otherwise the option is dynamic, sometimes exists and sometimes it doesn't. That looks like an anti-pattern to me (it's like setting a dynamic property in a PHP object).

Reply:
If you argue this way, it should not be part of the data because it's not exposed at all and is still dynamic. It should be a property of the class.

Reply:
Yes ... in strict PHP it would be wrong ... but in collectors we do it this way *(from what I've seen ... I can be wrong)*.

Reply:
Nvm only data is serialized.